### PR TITLE
feat(ME-2854): policy v2 support

### PR DESCRIPTION
**Pull request by btoonk**
# Description

policy v2 support
Fixes # (ME-2854)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

## Summary by CodeRabbit

- **New Features**
  - Enhanced permission management with new types: HTTP, TLS, VNC, RDP, and VPN permissions.
  - Improved user experience with updated database name input prompt.

- **Bug Fixes**
  - Restructured permissions to streamline and enhance security settings.

- **Refactor**
  - Simplified permissions structure for better clarity and maintainability.

### CODE REVIEW

File: internal/sqlauthproxy/mssql.go [hunk 1]
```diff
@@ -144,6 +144,23 @@ func (h mssqlHandler) handleClient(c net.Conn) {
...
+
+	if e2eeConn.Metadata == nil {
+		h.Logger.Error("invalid e2e metadata")
+		return
+	}
+
+	if !isDatabaseAllowed(e2eeConn.Metadata.AllowedActions, login.Database) {
+		h.Logger.Error("database access denied by policy", zap.String("database", login.Database), zap.String("user", e2eeConn.Metadata.UserEmail))
+		return
+	}
```
[the remark this answers is not in the document]
Comment (btoonk) [reply internal/sqlauthproxy/mssql.go:162]: I need e2eeConn later on.

File: internal/client/resources.go [hunk 2]
```diff
@@ -529,7 +529,7 @@ func EnterDBName(inputDBName, suggestedDBname string) (enteredDBName string, err
 	enteredDBName = inputDBName
 	if enteredDBName == "" {
 		if err = survey.AskOne(&survey.Input{
-			Message: "what is the name of the database schema:",
+			Message: "what is the name of the database database:",
```
[the remark this answers is not in the document]
Comment (btoonk) [reply internal/client/resources.go:532]: thanks

File: internal/api/api.go [hunk 3]
```diff
@@ -747,6 +738,61 @@ func (a *Border0API) UpdateSession(update models.SessionUpdate) (err error) {
 	return
 }
 
+func decodePermission(permission any, permissionType string) (models.Permissions, error) {
+	var permissions models.Permissions
+	var p any
+
+	switch permissionType {
+	case "ssh":
```
Comment (adrianosela) [inline internal/api/api.go:746]: there's constants for all this in the already-imported SDK. e.g. service.ServiceTypeSSH, etc

File: internal/api/models/policy.go [hunk 4]
```diff
@@ -82,70 +83,137 @@ type AddSocketToPolicyRequest struct {
...
+	AllowedContainers *[]string `json:"allowed_containers,omitempty"`
+}
+
+type HTTPPermissions struct{}
+type TLSPermissions struct{}
+type VNCPermissions struct{}
+type RDPPermissions struct{}
+type VPNPermissions struct{}
+
+// type Permissions struct {
```
Comment (adrianosela) [inline internal/api/models/policy.go:153]: delete dead code

**Review by adrianosela — APPROVED** [review]
LGTM. Make sure to test a socket of each protocol incl RDP, VNC, VPN before shipping. Can help with that if needed

File: internal/api/api.go (hunk 3 above)
Comment (btoonk) [reply internal/api/api.go]: fixed

File: internal/api/models/policy.go (hunk 4 above)
Comment (btoonk) [reply internal/api/models/policy.go]: fixed

**Pull request merged**